### ISSUE TITLE
no-wrap-func rule does not catch parenthesised FEs in certain positions

### ISSUE THREAD

**Issue by michaelficarra** [eslint/eslint#2466]
This program will trigger no warnings when the `no-wrap-func` rule is enabled, though I would expect one per line:

```js
a[(function(){})];
f((function(){}));
new f((function(){}));
```

**Comment by ilyavolodin** [comment]
@michaelficarra I think this rule was meant for function declarations/expression vs. invocation. It's been a long while since I wrote that rule, but based on documentation, the original idea was to disallow parentheses to provide visual difference between functional expression and IIFE.

**Comment by ilyavolodin** [comment]
Sorry ignore the comment above. I missed the first example, which is a bug no matter what.

**Comment by nzakas** [comment]
Thanks @michaelficarra

**Comment by nzakas** [comment]
Working on this.
